src/kaltura-player.js

```diff
@@ -510,6 +516,33 @@ class KalturaPlayer extends FakeEventTarget {
   get Error(): typeof Error {
     return this._localPlayer.Error;
   }
+
+  /**
+   * set the share config
+   * @returns {void}
+   * @private
+   */
+  _maybeSetEmbedConfig(): void {
+    const ui = this.config.ui;
+    if (ui && ui.components && ui.components.share) {
+      const share = this.config.ui.components.share;
+      let shareUrl = share.shareUrl;
+      let embedUrl = share.embedUrl;
+      const shareConfig = {
+        ui: {
+          components: {
+            share: {
+              ...(shareUrl && shareUrl),
+              ...(embedUrl && embedUrl)
```

Comment:
can you explain the purpose and the duplication?

Reply:
it's a way to define conditional if to add member to object.
It's the equivalent of writing:
```javascript
if (shareUrl) {
  share.shareUrl = shareUrl;
}
if (embedUrl) {
  share.embedUrl = embedUrl;
}
```

Reply:
for me `...{x && x}` does something else. bad thing:
```
x = 'Oren';
{...(x && x)} // ['O','r','e','n']
```